### ISSUE TITLE
https login problem

### ISSUE THREAD

**Issue by winterhard** [atauenis/webone#21]
Login problem with https phpBB forums (maybe others as well, haven't tried yet). Username and password not recognized. Using local mode.

**Comment by atauenis** [comment]
Yes, seems that proxy in local mode does not correctly editing cookies and URLs to local mode style. I'll try to fix this in nearest next (v0.10.5) release.

**Comment by atauenis** [comment]
As testing saw, the latest codebase from Git no longer have this bug.

**Comment by winterhard** [comment]
OK, so it will be included in v.0.10.5 release? Unfortunately, I cannot compile the code myself.

**Comment by atauenis** [comment]
Yes, I'm planning to release v0.10.5 in this month. And it will work correctly in local mode.

**Comment by winterhard** [comment]
Great! Thanks for your excellent work!

**Comment by atauenis** [comment]
Published new version 0.10.5, it have no problems on my known phpBB 3.x forums (tested on phantom.sannata.org and rt20.mybb2.ru). The only exception is VOGONS, where still happens problem on log in (the forum admin seems to have enabled a bot protection, which is detecting attempts to log in via proxy).

**Comment by winterhard** [comment]
Hm, I still cannot login to atari-forum.com (phpBB 3.x AFAIK). My browser doesn't know port numbers so I have the proxy running on port 80 (not 8080) and from the browser connect to http://127.0.0.1/http:/atari-forum.com. Everthings seems to work, except login. Could this be related to the port number?

**Comment by atauenis** [comment]
Registered on Atari Forum, checked, really, it saying `The submitted form was invalid. Try submitting again.` on correct username and password.

I'll investigate this. However, the forum really have a bot-protecting plugin. When I tried to register on atari-forum.com using my e-mail on yandex.ru, the forum rejected it (however Yandex Mail is a legal and popular mailbox service in ex. USSR). So to register on it I'm forced to use another e-mail address.

**Comment by winterhard** [comment]
I have contacted the admin fo the forum about this. I am definitely using the correct user/pw so it may have something to do with their protection. I will let you know as soon as I know more.

**Comment by atauenis** [comment]
I've found difference between forums where login work, and where don't. Forums where all is working are running on phpBB 3.2, and which not working are running on phpBB 3.3. Atari Forum is using phpBB 3.3 too.

Tested it in normal mode of proxy with Firefox 3.6.28, login working.
![AtariForum firefox 3 6](https://user-images.githubusercontent.com/4800266/107422311-eaae7680-6b2b-11eb-8ac7-fd49ebb039a8.png)

Then reopened it in latest Chrome, and found the error. PhpBB 3.3 is using `.example.com` domain for cookies, while phpBB 3.0-3.2 used `example.com`. I'd forget to implement patching this kind of cookies, so this is (probably) the reason why phpBB 3.3 forums are not usable with WebOne in local mode.
![AtariForum cookie](https://user-images.githubusercontent.com/4800266/107421730-622fd600-6b2b-11eb-997c-290499f24b89.png)

I'll fix this in next version.

**Comment by atauenis** [comment]
Seems, this is caused not only by the cookie domain. Fixed it, the alert in DevTools disappeared, but cookies still don't appear in Application tab of DevTools and the bug still present. So this problem will take a time to investigate and fix.

**Comment by winterhard** [comment]
Thanks for your efforts, I really appreciate it. It would be wonderful if you could get it to work.

**Comment by atauenis** [comment]
Huh, probably found the source of problem. Somewhere Set-Cookie HTTP header becames be in strange format, with multiple cookies per HTTP header. And browser is saving only first cookie. Now I am working on rewrite of cookie processing to get this problem out.

**Comment by winterhard** [comment]
Ah, looking forward to that. Just tried the latest 0.11.1 but the issue is still there. Take your time.

**Comment by atauenis** [comment]
Yes, I'm fixed Set-Cookie header format, this solved some other problems, but not this. And I'm stumped again.
